packages/js/product-editor/src/components/blocks-template/blocks-template.tsx

```diff
@@ -0,0 +1,38 @@
+/**
+ * External dependencies
+ */
+import { useSelect } from '@wordpress/data';
+import { store as blockEditorStore } from '@wordpress/block-editor';
+import { synchronizeBlocksWithTemplate } from '@wordpress/blocks';
+import { useLayoutEffect } from '@wordpress/element';
+import {
+	// eslint-disable-next-line @typescript-eslint/ban-ts-comment
+	// @ts-ignore
+	useEntityBlockEditor,
+	// eslint-disable-next-line @typescript-eslint/ban-ts-comment
+	// @ts-ignore
+	useEntityId,
+} from '@wordpress/core-data';
+
+export function BlocksTemplate() {
+	const blockEditorSettings = useSelect( ( select ) => {
+		// eslint-disable-next-line @typescript-eslint/ban-ts-comment
+		// @ts-ignore
+		return select( blockEditorStore ).getSettings();
+	}, [] );
+
+	const productId = useEntityId( 'postType', 'product' );
+
+	const [ , , onChange ] = useEntityBlockEditor( 'postType', 'product', {
+		id: productId,
+	} );
+
+	useLayoutEffect( () => {
+		onChange(
+			synchronizeBlocksWithTemplate( [], blockEditorSettings?.template ),
+			{}
+		);
+	}, [ blockEditorSettings?.template ] );
+
+	return null;
```

Comment:
This null render feels odd. It seems like at heart this `BlocksTemplate` is more of a hook? Is it done this way because it has to exist under `BlockContextProvider` and/or `BlockEditorProvider` and we don't have a component function to do that from?

Reply:
Ya, it does feel a bit odd.

> Is it done this way because it has to exist under BlockContextProvider and/or BlockEditorProvider and we don't have a component function to do that from?

Exactly.